### ISSUE TITLE
GitHub action workflow uses the wrong PyPI token

### ISSUE THREAD

**Issue by billsioros** [relekang/python-semantic-release#335]
## The problem

Whilst using the module via GitHub Actions, `twine` fails to upload to PyPI

## Expected behavior

I expected a new version of the module being uploaded to PyPI

## Additional context

<details>
  <summary>My GitHub Action workflow!</summary>

  ```yml
  name: Publish

on:
    pull_request:
        branches:
            - master
        types:
            - closed

jobs:
    publish:
        name: Publish
        runs-on: ubuntu-latest
        if: github.event.pull_request.merged == true
        steps:
            - uses: actions/checkout@v2
              with:
                  fetch-depth: 0

            - name: Create a GitHub Release
              uses: relekang/python-semantic-release@v7.15.4
              with:
                  github_token: ${{ secrets.GITHUB_TOKEN }}
                  pypi_token: ${{ secrets.DOTIFY_TOKEN }}
```

</details>

<details>
  <summary>GitHub workflow logs</summary>
  Run relekang/python-semantic-release@v7.15.4
<pre><code>
/usr/bin/docker run --name a33c12253e46339a44d32bb6b368a5e0f3f6c_53c9ba --label 8a33c1 --workdir /github/workspace --rm -e pythonLocation -e LD_LIBRARY_PATH -e INPUT_GITHUB_TOKEN -e INPUT_PYPI_TOKEN -e INPUT_DIRECTORY -e INPUT_PYPI_USERNAME -e INPUT_PYPI_PASSWORD -e HOME -e GITHUB_JOB -e GITHUB_REF -e GITHUB_SHA -e GITHUB_REPOSITORY -e GITHUB_REPOSITORY_OWNER -e GITHUB_RUN_ID -e GITHUB_RUN_NUMBER -e GITHUB_RETENTION_DAYS -e GITHUB_ACTOR -e GITHUB_WORKFLOW -e GITHUB_HEAD_REF -e GITHUB_BASE_REF -e GITHUB_EVENT_NAME -e GITHUB_SERVER_URL -e GITHUB_API_URL -e GITHUB_GRAPHQL_URL -e GITHUB_WORKSPACE -e GITHUB_ACTION -e GITHUB_EVENT_PATH -e GITHUB_ACTION_REPOSITORY -e GITHUB_ACTION_REF -e GITHUB_PATH -e GITHUB_ENV -e RUNNER_OS -e RUNNER_TOOL_CACHE -e RUNNER_TEMP -e RUNNER_WORKSPACE -e ACTIONS_RUNTIME_URL -e ACTIONS_RUNTIME_TOKEN -e ACTIONS_CACHE_URL -e GITHUB_ACTIONS=true -e CI=true -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" -v "/home/runner/work/_temp/_runner_file_commands":"/github/file_commands" -v "/home/runner/work/dotify/dotify":"/github/workspace" 8a33c1:2253e46339a44d32bb6b368a5e0f3f6c
/semantic-release/action.sh: line 3: set: -l: invalid option
set: usage: set [-abefhkmnptuvxBCHP] [-o option-name] [--] [arg ...]
debug: get_current_version_by_config_file()
debug: Regex matched version: 0.3.6
debug: get_current_version_by_config_file -> 0.3.6
debug: evaluate_version_bump(0.3.6, None)
debug: parse_commit_message('Merge pull request #39 from billsioros/development
debug: 
debug: fix(docs.yml): running `mkdocs` via poetry')
debug: Ignoring UnknownCommitMessageStyleError: Unable to parse the given commit message: Merge pull request #39 from billsioros/development
debug: 
debug: fix(docs.yml): running `mkdocs` via poetry
debug: parse_commit_message('fix(docs.yml): running `mkdocs` via poetry')
debug: parse_commit_message -> ParsedCommit(bump=1, type='fix', scope='docs.yml', descriptions=['running `mkdocs` via poetry'], breaking_descriptions=[])
debug: Commits found since last release: 2
debug: evaluate_version_bump -> patch
debug: get_new_version(0.3.6, 'patch')
debug: get_new_version -> 0.3.7
debug: get_repository_owner_and_name()
debug: get_repository_owner_and_name -> ('billsioros', 'dotify')
debug: Running publish on branch master
debug: checkout('master')
debug: checkout -> Your branch is up to date with 'origin/master'.
debug: generate_changelog(0.3.6)
debug: parse_commit_message('Merge pull request #39 from billsioros/development
debug: 
debug: fix(docs.yml): running `mkdocs` via poetry')
debug: Ignoring UnknownCommitMessageStyleError: Unable to parse the given commit message: Merge pull request #39 from billsioros/development
debug: 
debug: fix(docs.yml): running `mkdocs` via poetry
debug: parse_commit_message('fix(docs.yml): running `mkdocs` via poetry')
debug: parse_commit_message -> ParsedCommit(bump=1, type='fix', scope='docs.yml', descriptions=['running `mkdocs` via poetry'], breaking_descriptions=[])
debug: Creating new changelog section for fix 
debug: generate_changelog -> {'breaking': [], 'fix': [('9475e7322cae23e66acb895d1fd471dc7f161eba', '**docs.yml:** Running `mkdocs` via poetry')]}
debug: markdown_changelog('billsioros', 'dotify', '0.3.7', {'breaking': [], 'fix': [('9475e7322cae23e66acb895d1fd471dc7f161eba', '**docs.yml:** Running `mkdocs` via poetry')]}, header=False, previous_version=0.3.6)
debug: get_repository_owner_and_name()
debug: get_repository_owner_and_name -> ('billsioros', 'dotify')
debug: markdown_changelog -> ### Fix
debug: * **docs.yml:** Running `mkdocs` via poetry ([`9475e73`](https://github.com/billsioros/dotify/commit/9475e7322cae23e66acb895d1fd471dc7f161eba))
debug: 
debug: **[See all commits in this version](https://github.com/billsioros/dotify/compare/v0.3.6...v0.3.7)**
debug: update_changelog_file('0.3.7', '### Fix
debug: * **docs.yml:** Running `mkdocs` via poetry ([`9475e73`](https://github.com/billsioros/dotify/commit/9475e7322cae23e66acb895d1fd471dc7f161eba))
debug: 
debug: **[See all commits in this version](https://github.com/billsioros/dotify/compare/v0.3.6...v0.3.7)**')
debug: set_new_version('0.3.7')
debug: set_new_version -> True
debug: commit_new_version('0.3.7')
debug: commit_new_version -> [master 87faf54] 0.3.7
debug:  2 files changed, 7 insertions(+), 1 deletion(-)
debug: tag_new_version('0.3.7')
debug: tag_new_version -> 
Bumping with a patch version to 0.3.7
Pushing new version
debug: get_hvcs()
debug: get_hvcs -> <class 'semantic_release.hvcs.Github'>
debug: get_hvcs()
debug: get_hvcs -> <class 'semantic_release.hvcs.Github'>
debug: push_new_version(, auth_token='***', owner='billsioros', name='dotify', branch='master', domain='github.com')
Building distributions
debug: Running rm -rf dist
Running python -m pip install poetry && poetry build
WARNING: The directory '/github/home/.cache/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you should use sudo's -H flag.
WARNING: Value for scheme.headers does not match. Please report this to <https://github.com/pypa/pip/issues/9617>
distutils: /usr/local/include/python3.7m/UNKNOWN
sysconfig: /usr/local/include/python3.7m
WARNING: Additional context:
user = False
home = None
root = None
prefix = None
Collecting poetry
  Downloading poetry-1.1.6-py2.py3-none-any.whl (172 kB)
Collecting requests-toolbelt<0.10.0,>=0.9.1
  Downloading requests_toolbelt-0.9.1-py2.py3-none-any.whl (54 kB)
Collecting virtualenv<21.0.0,>=20.0.26
  Downloading virtualenv-20.4.4-py2.py3-none-any.whl (7.2 MB)
Collecting cachecontrol[filecache]<0.13.0,>=0.12.4
  Downloading CacheControl-0.12.6-py2.py3-none-any.whl (19 kB)
Collecting packaging<21.0,>=20.4
  Downloading packaging-20.9-py2.py3-none-any.whl (40 kB)
Collecting html5lib<2.0,>=1.0
  Downloading html5lib-1.1-py2.py3-none-any.whl (112 kB)
Collecting pkginfo<2.0,>=1.4
  Downloading pkginfo-1.7.0-py2.py3-none-any.whl (25 kB)
Collecting poetry-core<1.1.0,>=1.0.3
  Downloading poetry_core-1.0.3-py2.py3-none-any.whl (424 kB)
Collecting shellingham<2.0,>=1.1
  Downloading shellingham-1.4.0-py2.py3-none-any.whl (9.4 kB)
Collecting requests<3.0,>=2.18
  Downloading requests-2.25.1-py2.py3-none-any.whl (61 kB)
Collecting cachy<0.4.0,>=0.3.0
  Downloading cachy-0.3.0-py2.py3-none-any.whl (20 kB)
Collecting clikit<0.7.0,>=0.6.2
  Downloading clikit-0.6.2-py2.py3-none-any.whl (91 kB)
Collecting tomlkit<1.0.0,>=0.7.0
  Downloading tomlkit-0.7.0-py2.py3-none-any.whl (32 kB)
Collecting crashtest<0.4.0,>=0.3.0
  Downloading crashtest-0.3.1-py3-none-any.whl (7.0 kB)
Collecting keyring<22.0.0,>=21.2.0
  Downloading keyring-21.8.0-py3-none-any.whl (32 kB)
Collecting importlib-metadata<2.0.0,>=1.6.0
  Downloading importlib_metadata-1.7.0-py2.py3-none-any.whl (31 kB)
Collecting cleo<0.9.0,>=0.8.1
  Downloading cleo-0.8.1-py2.py3-none-any.whl (21 kB)
Collecting pexpect<5.0.0,>=4.7.0
  Downloading pexpect-4.8.0-py2.py3-none-any.whl (59 kB)
Collecting msgpack>=0.5.2
  Downloading msgpack-1.0.2-cp37-cp37m-manylinux1_x86_64.whl (273 kB)
Collecting lockfile>=0.9
  Downloading lockfile-0.12.2-py2.py3-none-any.whl (13 kB)
Collecting pylev<2.0,>=1.3
  Downloading pylev-1.3.0-py2.py3-none-any.whl (4.9 kB)
Collecting pastel<0.3.0,>=0.2.0
  Downloading pastel-0.2.1-py2.py3-none-any.whl (6.0 kB)
Collecting six>=1.9
  Downloading six-1.15.0-py2.py3-none-any.whl (10 kB)
Collecting webencodings
  Downloading webencodings-0.5.1-py2.py3-none-any.whl (11 kB)
Collecting zipp>=0.5
  Downloading zipp-3.4.1-py3-none-any.whl (5.2 kB)
Collecting jeepney>=0.4.2
  Downloading jeepney-0.6.0-py3-none-any.whl (45 kB)
Collecting SecretStorage>=3.2
  Downloading SecretStorage-3.3.1-py3-none-any.whl (15 kB)
Collecting pyparsing>=2.0.2
  Downloading pyparsing-2.4.7-py2.py3-none-any.whl (67 kB)
Collecting ptyprocess>=0.5
  Downloading ptyprocess-0.7.0-py2.py3-none-any.whl (13 kB)
Collecting urllib3<1.27,>=1.21.1
  Downloading urllib3-1.26.4-py2.py3-none-any.whl (153 kB)
Collecting idna<3,>=2.5
  Downloading idna-2.10-py2.py3-none-any.whl (58 kB)
Collecting certifi>=2017.4.17
  Downloading certifi-2020.12.5-py2.py3-none-any.whl (147 kB)
Collecting chardet<5,>=3.0.2
  Downloading chardet-4.0.0-py2.py3-none-any.whl (178 kB)
Collecting cryptography>=2.0
  Downloading cryptography-3.4.7-cp36-abi3-manylinux2014_x86_64.whl (3.2 MB)
Collecting cffi>=1.12
  Downloading cffi-1.14.5-cp37-cp37m-manylinux1_x86_64.whl (402 kB)
Collecting pycparser
  Downloading pycparser-2.20-py2.py3-none-any.whl (112 kB)
Collecting filelock<4,>=3.0.0
  Downloading filelock-3.0.12-py3-none-any.whl (7.6 kB)
Collecting distlib<1,>=0.3.1
  Downloading distlib-0.3.1-py2.py3-none-any.whl (335 kB)
Collecting appdirs<2,>=1.4.3
  Downloading appdirs-1.4.4-py2.py3-none-any.whl (9.6 kB)
Installing collected packages: pycparser, urllib3, idna, chardet, cffi, certifi, zipp, requests, pylev, pastel, msgpack, jeepney, cryptography, crashtest, webencodings, six, SecretStorage, pyparsing, ptyprocess, lockfile, importlib-metadata, filelock, distlib, clikit, cachecontrol, appdirs, virtualenv, tomlkit, shellingham, requests-toolbelt, poetry-core, pkginfo, pexpect, packaging, keyring, html5lib, cleo, cachy, poetry
WARNING: Value for scheme.headers does not match. Please report this to <https://github.com/pypa/pip/issues/9617>
distutils: /usr/local/include/python3.7m/UNKNOWN
sysconfig: /usr/local/include/python3.7m
WARNING: Additional context:
user = False
home = None
root = None
prefix = None
WARNING: Running pip as root will break packages and permissions. You should install packages reliably by using venv: https://pip.pypa.io/warnings/venv
Successfully installed SecretStorage-3.3.1 appdirs-1.4.4 cachecontrol-0.12.6 cachy-0.3.0 certifi-2020.12.5 cffi-1.14.5 chardet-4.0.0 cleo-0.8.1 clikit-0.6.2 crashtest-0.3.1 cryptography-3.4.7 distlib-0.3.1 filelock-3.0.12 html5lib-1.1 idna-2.10 importlib-metadata-1.7.0 jeepney-0.6.0 keyring-21.8.0 lockfile-0.12.2 msgpack-1.0.2 packaging-20.9 pastel-0.2.1 pexpect-4.8.0 pkginfo-1.7.0 poetry-1.1.6 poetry-core-1.0.3 ptyprocess-0.7.0 pycparser-2.20 pylev-1.3.0 pyparsing-2.4.7 requests-2.25.1 requests-toolbelt-0.9.1 shellingham-1.4.0 six-1.15.0 tomlkit-0.7.0 urllib3-1.26.4 virtualenv-20.4.4 webencodings-0.5.1 zipp-3.4.1
WARNING: You are using pip version 21.1; however, version 21.1.1 is available.
You should consider upgrading via the '/usr/local/bin/python -m pip install --upgrade pip' command.
Creating virtualenv dotify-cDQ_M3T8-py3.7 in /github/home/.cache/pypoetry/virtualenvs
Building dotify (0.3.7)
  - Building sdist
  - Built dotify-0.3.7.tar.gz
  - Building wheel
  - Built dotify-0.3.7-py3-none-any.whl
Uploading to PyPI
debug: upload_to_pypi(, path='dist', skip_existing=False, glob_patterns=None)
Uploading distributions to https://upload.pypi.org/legacy/
Uploading dotify-0.3.7-py3-none-any.whl

  0%|          | 0.00/22.8k [00:00<?, ?B/s]
 35%|███▌      | 8.00k/22.8k [00:00<00:00, 30.8kB/s]
100%|██████████| 22.8k/22.8k [00:00<00:00, 46.3kB/s]
NOTE: Try --verbose to see response content.
HTTPError: 403 Forbidden from https://upload.pypi.org/legacy/
Invalid or non-existent authentication information. See https://pypi.org/help/#invalid-auth for more information.
error: Encountered a bad command exit code!
error: 
error: Command: 'twine upload -u \'__token__\' -p \'***\n\' "dist/*"'
error: 
error: Exit code: 1
error: 
error: Stdout: already printed
error: 
error: Stderr: already printed
error: 
</code></pre>
</details>

TLDR; For some reason I can spot a newline has cripped into the command `'twine upload -u \'__token__\' -p \'***\n\' "dist/*"'`. I'm not really confident, when saying this, but I have already gone through the issues and the documentation and I couldn't spot any mentions of this peculiar issue.

Let me also clarify that the token is valid and has been tested via other GitHub actions, such as `JRubics/poetry-publish@v1`.

**Comment by billsioros** [comment]
If I understand correctly, the module does not strip trailing whitespace from the token. I have tackled the issue by simply overwriting my existing PyPI token.
